tests/files.cpp

```diff
@@ -142,6 +146,16 @@ int main()
   FILE *n = fopen("/dev/null", "w");
   printf("5 bytes to dev/null: %zu\n", fwrite(data, 1, 5, n));
   fclose(n);
+  
+  // Test file creation (regression test for #16784)
+  const char *target_file = "test.out";
+  int target_fd;
+  remove(target_file);
+  target_fd = openat(AT_FDCWD, target_file, O_WRONLY | O_CREAT | O_TRUNC | O_CLOEXEC, 33060);
```

Comment:
I think you can just do `open(target_file` to keep to simple.   Also, I think can you drop the `O_CLOEXEC` and the mode without effecting the test.   

Also since this test is running against an ephemeral MEMFS you can drop the `remove` and just do `open("test.out"...`

Reply:
Omitting the mode when using `open` to create a new file is UB:

```
The mode argument must be supplied if O_CREAT or O_TMPFILE is
specified in flags; if it is not supplied, some arbitrary
bytes from the stack will be applied as the file mode.
- https://man7.org/linux/man-pages/man2/open.2.html
```

Reply:
OK, can use the S_ macros (e.g. S_IRUSR) rather than a hardcoded constant then?